update and clean up migration docs (#18527)
cockroachdb/docs: src/current/v24.1/migrate-from-mysql.md

       Click **Save**.

>        This is a workaround to prevent [data validation](#step-3-validate-the-migrated-data) from failing due to collation mismatches. For more details, see the [MOLT Verify] ({% link {{ page.version.version }}/molt-verify.md %}#known-limitations) documentation.

Reviewer: ```suggestion
       This workaround helps to prevent [data validation](#step-3-validate-the-migrated-data) from failing due to collation mismatches. For more details, refer to [MOLT Verify Known Limitations] ({% link {{ page.version.version }}/molt-verify.md %}#known-limitations).
```
